patroni/ha.py

```diff
@@ -730,13 +730,19 @@ def _run_cycle(self):
             if self._async_executor.busy:
                 return self.handle_long_action_in_progress()
 
-            # we've got here, so any async action has finished. Check if we tried to recover and failed
+            # we've got here, so any async action has finished.
             if self.recovering and not self.state_handler.need_rewind:
                 self.recovering = False
+                # Check if we tried to recover and failed
                 msg = self.post_recover()
                 if msg is not None:
                     return msg
 
+                if not self.has_lock() and len(self.cluster.members) == 1 and \
+                        not (self.is_synchronous_mode() and self.cluster.sync and
+                             self.cluster.sync.matches(self.state_handler.name)):
+                    return 'too few members in cluster after "recovery", postponing leader race'
+
```

Comment:
I was just trying not to change much existing behavior. I.e. on the first iteration it was starting postgres and only on the next iteration (after loop_wait) it was doing reader race and promotion.

Such delay might be useful only in one case: if the cluster is completely down and you are bringing it back (all nodes more or less at the same time).